hiku/result.py

```diff
@@ -71,8 +71,20 @@ def ref(self, node, ident):
         return Ref(self.index, node, ident)
 
 
-def _filter_fields(result, node):
-    return {f.name: result[f.name] for f in node.fields}
+def _denormalize_type(type_, result, query_node):
+    if isinstance(type_, SequenceMeta):
+        return [_denormalize_type(type_.__item_type__, item, query_node)
+                for item in result]
+    elif isinstance(type_, OptionalMeta):
+        return (_denormalize_type(type_.__type__, result, query_node)
+                if result is not None else None)
+    elif isinstance(type_, RecordMeta):
+        field_types = type_.__field_types__
+        return {f.name: _denormalize_type(field_types[f.name], result[f.name],
+                                          getattr(f, 'node', f))
```

Comment:
`getattr(f, 'node', f)` looks very unpleasant. I think we can get rid of it, if `_denormalize_type` function will accept more general argument type than `query_node` – `query_obj`. In case of `Record` this argument should be an instance of `query.Link`, in case of scalar type, this argument should be an instance of `query.Field`.

Reply:
Actually, query_obj, passed to this function initially will always be of type `Link`. See [`hiku/result.py#L99`](https://github.com/vmagamedov/hiku/pull/32/files#diff-094ccce2936cab0673248d5971ebbcafR99) for reference.

As for `getattr(f, 'node', f)` loking very unpleasant I agree. Will try to come up with solution which does not uses such kind of construction.

Reply:
Fixed.